clippy_lints/src/non_copy_const.rs

```diff
@@ -34,6 +34,14 @@ declare_clippy_lint! {
     /// `std::sync::ONCE_INIT` constant). In this case the use of `const` is legit,
     /// and this lint should be suppressed.
     ///
+    /// When an enum has variants with interior mutability, use of its non interior mutable
+    /// variants can generate false positives. See issue [#3962](https://github.com/rust-lang/rust-clippy/issues/3962)
+    ///
+    /// Types that have underlying or potential interior mutability trigger the lint, but are
+    /// usually not the type of errors that the lint is trying to catch. See issues
+    /// [#5050](https://github.com/rust-lang/rust-clippy/issues/5050),
+    /// [#5812](https://github.com/rust-lang/rust-clippy/issues/5812)
```

Comment:
About #5812, I'm not sure TBH. Reading the issue thread, it would seem to me that the lint is right in this case.

cc @flip1995

Reply:
The lint is wrong to lint in #5812, but it is impossible for Clippy to check, if the interior mutable part of a struct is actually used somewhere, or just there for some other reasons. So mentioning this issue here is correct IMO.